app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -0,0 +1,175 @@
+        Log.d(TAG, "setCategory: ");
+        this.category = category;
+    }
+
+    @Override
+    public Category getCategory() {
+        return this.category;
+    }
+
+    @Override
+    public void setShowFavorite(Boolean showOnlyFavorite) {
+        Log.d(TAG, "setShowFavorite: " + showOnlyFavorite);
+        this.showOnlyFavorite = showOnlyFavorite;
+    }
+
+    @Override
+    public Boolean getShowFavorite() {
+        return this.showOnlyFavorite;
+    }
+
+    @Override
+    public void invertFavoriteState() {
+        showOnlyFavorite = !showOnlyFavorite;
+        Log.d(TAG, "invertFavoriteState: " + showOnlyFavorite);
+        view.setFavoriteIcon(showOnlyFavorite);
+        loadNewsFromDB();
+        view.updateNewsListUI(news);
+    }
+
+    @Override
+    public void updateNewsFromApi() {
+        Log.d(TAG, "updateNewsFromApi: ");
+        loadNewsFromApi();
+    }
+
+    private void showNewsInBrowserByUrl(int position) {
+        NewsItem item = news.get(position);
+        String url = item.getUrl();
+        view.showNewsDetailsUI(url);
+    }
+
+    private List<NewsItem> getNewsFromDB() {
+        return newsStorage.getNewsFromDB(showOnlyFavorite, category);
+    }
+
+    private void initNewsStorage() {
+        NewsItemDAO newsItemDAO = mainApplication.getNewsDataBase().newsItemDAO();
+        newsStorage = new NewsStorage(newsItemDAO);
+    }
+
+    private void initNewsItemListCallback() {
+        // TODO: 17.03.2020 extract in network layer
+        callbackNewsItemList = new Callback<NewsApiResponse>() {
+            @Override
+            public void onResponse(@NotNull Call<NewsApiResponse> call, @NotNull Response<NewsApiResponse> response) {
+                view.hideProgressBar();
+                if (!response.isSuccessful()) {
+                    Log.d(TAG, "onResponse " + response.code());
+                    return;
+                }
+                newsStorage.insertUnique(getNewsItemListByResponse(response, category));
+                Log.d(TAG, "onResponse: ");
+                loadNewsFromDB();
+                view.updateNewsListUI(news);
+            }
+
+            @Override
+            public void onFailure(@NotNull Call<NewsApiResponse> call, @NotNull Throwable t) {
+                view.hideProgressBar();
+                Log.d(TAG, "onFailure " + Objects.requireNonNull(t.getMessage()));
+            }
+        };
+    }
+
+    private void performCall() {
+        Log.d(TAG, "performCall: ");
+        NewsApiRepository newsApiRepository = new NewsApiRepository(mainApplication);
+        newsApiRepository.getNewsFromApi(category, callbackNewsItemList);
+    }
+
+    private List<NewsItem> getNewsItemListByResponse(Response<NewsApiResponse> response, Category category) {
+        List<NewsItem> news = new ArrayList<>();
+        List<NewsApiResponseItem> newsApiResponseItems = Objects.requireNonNull(response.body()).getNewsApiResponseItemList();
```

Comment:
The same here - we can make things simpler by calling a variable `responseItems`.